macros/stitch/base/stitch_fb_ad_creatives.sql

```diff
@@ -9,38 +9,46 @@
 
```

Comment:
With this model we should be able to remove the Snowflake version altogether, right? The whole point of our efforts on the cross-database functions was to make it possible to write the same SQL for each adapter, so we should need to maintain multiple implementations of this model given that, right?

Do you see any other reasons why we need multiple implementations of this model?

Reply:
the first CTE `base` have different syntaxes for the 'default' and 'snowflake' implementations:

the default implementation looks like this where the code for creating the URL field looks like this:
```
    select
    
        id,
        lower(nullif(url_tags, '')) as url_tags,
        lower(coalesce(
          nullif(object_story_spec__link_data__call_to_action__value__link, ''),
          nullif(object_story_spec__video_data__call_to_action__value__link, ''),
          nullif(object_story_spec__link_data__link, '')
        )) as url
```

and then the 'snowflake' implementation has brackets:
```
with base as (

    select

        id,
        lower(coalesce(
          nullif(object_story_spec['link_data']['call_to_action']['value']['link']::varchar, ''),
          nullif(object_story_spec['video_data']['call_to_action']['value']['link']::varchar, ''),
          nullif(object_story_spec['link_data']['link']::varchar, '')
        )) as url
```

i've never used the facebook package on a client that used Snowflake, so i'm not sure what the fields are called, but it looks like the fields in this model are named differently for Snowflake specifically, idk